pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -284,12 +290,48 @@ func validateParameterInOneOfMatrixOrParams(matrix []Param, params []Param) (err
 	return errs
 }
 
-func validateStringVariable(value, prefix string, stringVars sets.String, arrayVars sets.String) *apis.FieldError {
+// validateStringVariableP validates param variable usage in string value
+// This will allow referencing the whole object/array value. example:
+// params:
+// 	- name: arg
+// 		value: $(params.myObject[*])
+// see https://github.com/tektoncd/pipeline/issues/4879#issuecomment-1133120522 for background
+func validateStringVariableP(param Param, prefix string, paramNames sets.String, arrayVars sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
+	stringValue := param.Value.StringVal
+	singleVariableRegex := regexp.MustCompile(fmt.Sprintf("^\\$\\(%s\\.[_a-zA-Z][_a-zA-Z0-9.-]*\\[\\*\\]\\)$", prefix))
```

Comment:
Question: Don't we have a regex or helper methods we could use for this instead of another regex here?